File: c++/2.cpp
#include <iostream>
#include <fstream>
#include <sstream>
#include <string>

using std::string;

void usage(){
	std::cout << "Usage: main <word> <filename>\n";
}

int wordOccurrence(const string &word, const string &filename){
	std::ifstream file(filename);
	if (!file.is_open()){
		std::cerr << "Failed to open file " << filename << '\n';
		return -1;
	}

	std::stringstream buf;
	buf << file.rdbuf();

	int ret = 0;
	string split = "";
	for (char c : buf.str()){
		if (c == ' ' || c == '\t' || c == '\r' || c == '\n'){ // :skull:
			ret += split == word;

			split = "";
			continue;
		}

		split += c;
	}

	return ret;
}

int main(int argc, char *argv[]){
	if (argc < 3){
		usage();
		return 1;
	}

	std::cout << "\033[36m" << wordOccurrence(argv[1], argv[2]) << "\033[0m\n";
}

File: python/2.py
import sys

def usage():
	print("Usage: python3 2.py <filename>")

def wordOccurrence(filename):
	f = open(filename, "r")

	ret = 0
	for line in f:
		for char in line:
			if char == 'h':
				continue

			if char == 'i':
				ret += 1

	return ret

if len(sys.argv) < 2:
	usage()
	sys.exit(1)

print("\033[36m" + str(wordOccurrence(sys.argv[1])) + "\033[0m")
